lib/Analysis/Allocation.cpp

```diff
@@ -587,10 +585,23 @@ class AllocationAnalysis {
         Interval ySizeRange = {yStart, yStart + ySize};
         auto xOpRange = bufferRange.lookup(x);
         auto yOpRange = bufferRange.lookup(y);
+
+        // Buffers interfere if their allocation offsets overlap and they are
+        // live at the same time.
         if (xOpRange.intersects(yOpRange) &&
             xSizeRange.intersects(ySizeRange)) {
           interference[x].insert(y);
         }
+
+        // Buffers also interfer if their allocation offsets overlap and they
+        // exist within regions that may execute simultaneously.
+        auto wsx = x->id->getParentOfType<gpu::WarpSpecializeOp>();
```

Comment:
Is it possible to use any trait to express the "simultaneous" effect?
Just don't want to be very tight to this op

Reply:
I could try to add an interface or trait here.